### ISSUE TITLE
Errors on compiling sfemovie with libass

### ISSUE THREAD

**Issue by crazybytes** [Yalir/sfeMovie#109]
What libass should i use.
I cross compiled the ffmpeg-2.8 with ubuntu shell under windows. 
I did that before, with shared libs for ffmpeg.

Now i compiled it with the libass enabled. 
With help of the ffmpeg-windows-build-helpers-master.zip
It downloads and cross compiles all needed packages, als libass.

Then i use cmake-gui for windows, to set the sfemovie up with the compiled libs from ffmpeg, sfml, and the needed libs for subtitle support, and of course the libass.a en include files.
Normally it compiles with the codeblocks mingw, but this time i get errors.

[ 61%] Building CXX object CMakeFiles/sfeMovie.dir/src/VideoStream.cpp.obj
[ 66%] Linking CXX static library bin\libsfeMovie-s.a
[ 66%] Built target sfeMovie
Scanning dependencies of target sfeMovieDemo
[ 71%] Building CXX object examples/Demo/CMakeFiles/sfeMovieDemo.dir/main.cpp.obj
[ 76%] Building CXX object examples/Demo/CMakeFiles/sfeMovieDemo.dir/MediaInfo.cpp.obj
[ 80%] Building CXX object examples/Demo/CMakeFiles/sfeMovieDemo.dir/StreamSelector.cpp.obj
[ 85%] Building CXX object examples/Demo/CMakeFiles/sfeMovieDemo.dir/UserInterface.cpp.obj
[ 90%] Linking CXX executable ..\..\bin\sfeMovieDemo.exe
../../bin/libsfeMovie-s.a(SubtitleStream.cpp.obj):SubtitleStream.cpp:(.text+0xfd): undefined reference to `ass_set_frame_size'
../../bin/libsfeMovie-s.a(SubtitleStream.cpp.obj):SubtitleStream.cpp:(.text+0x187): undefined reference to `__ms_vsnprintf'
../../bin/libsfeMovie-s.a(SubtitleStream.cpp.obj):SubtitleStream.cpp:(.text+0xa3d): undefined reference to `ass_free_track'
../../bin/libsfeMovie-s.a(SubtitleStream.cpp.obj):SubtitleStream.cpp:(.text+0xc1f): undefined reference to `ass_new_track'
../../bin/libsfeMovie-s.a(SubtitleStream.cpp.obj):SubtitleStream.cpp:(.text+0xc5b): undefined reference to `ass_process_codec_private'
../../bin/libsfeMovie-s.a(SubtitleStream.cpp.obj):SubtitleStream.cpp:(.text+0xc8c): undefined reference to `ass_library_init'
../../bin/libsfeMovie-s.a(SubtitleStream.cpp.obj):SubtitleStream.cpp:(.text+0xc99): undefined reference to `ass_renderer_init'
../../bin/libsfeMovie-s.a(SubtitleStream.cpp.obj):SubtitleStream.cpp:(.text+0xcd0): undefined reference to `ass_set_message_cb'
../../bin/libsfeMovie-s.a(SubtitleStream.cpp.obj):SubtitleStream.cpp:(.text+0xd06): undefined reference to `ass_set_fonts'
../../bin/libsfeMovie-s.a(SubtitleStream.cpp.obj):SubtitleStream.cpp:(.text+0xd34): undefined reference to `ass_set_margins'
../../bin/libsfeMovie-s.a(SubtitleStream.cpp.obj):SubtitleStream.cpp:(.text+0xee0): undefined reference to `ass_renderer_done'
../../bin/libsfeMovie-s.a(SubtitleStream.cpp.obj):SubtitleStream.cpp:(.text+0xeed): undefined reference to `ass_library_done'
../../bin/libsfeMovie-s.a(SubtitleStream.cpp.obj):SubtitleStream.cpp:(.text+0x11b9): undefined reference to `ass_process_data'
../../bin/libsfeMovie-s.a(SubtitleStream.cpp.obj):SubtitleStream.cpp:(.text+0x28d2): undefined reference to `ass_render_frame'
../../bin/libsfeMovie-s.a(SubtitleStream.cpp.obj):SubtitleStream.cpp:(.text+0x3365): undefined reference to `ass_flush_events'
C:/CodeBlocks/MinGW/bin/../lib/gcc/mingw32/5.1.0/../../../../mingw32/bin/ld.exe: ../../bin/libsfeMovie-s.a(SubtitleStream.cpp.obj): bad reloc address 0x7 in section `.text$_ZNSt15_Sp_counted_ptrIPN3sfe10ASSLibraryELN9__gnu_cxx12_Lock_policyE2EED0Ev[__ZNSt15_Sp_counted_ptrIPN3sfe10ASSLibraryELN9__gnu_cxx12_Lock_policyE2EED0Ev]'
C:/CodeBlocks/MinGW/bin/../lib/gcc/mingw32/5.1.0/../../../../mingw32/bin/ld.exe: final link failed: Invalid operation
collect2.exe: error: ld returned 1 exit status
examples\Demo\CMakeFiles\sfeMovieDemo.dir\build.make:162: recipe for target 'bin/sfeMovieDemo.exe' failed
mingw32-make[2]: *** [bin/sfeMovieDemo.exe] Error 1
CMakeFiles\Makefile2:165: recipe for target 'examples/Demo/CMakeFiles/sfeMovieDemo.dir/all' failed
mingw32-make[1]: *** [examples/Demo/CMakeFiles/sfeMovieDemo.dir/all] Error 2
Makefile:150: recipe for target 'all' failed
mingw32-make: *** [all] Error 2

When i try to link another libass.a and include dir, or i set the libass wrong in cmake-gui, ii get errors on the first file it compiles:

D:\work\sfeMovie-master\out>mingw32-make
[  4%] Building CXX object CMakeFiles/sfeMovie.dir/src/AudioStream.cpp.obj
D:\work\sfeMovie-master\src\AudioStream.cpp:27:34: fatal error: libavformat/avformat.h: No such file or directory
compilation terminated.
CMakeFiles\sfeMovie.dir\build.make:62: recipe for target 'CMakeFiles/sfeMovie.dir/src/AudioStream.cpp.obj' failed
mingw32-make[2]: *** [CMakeFiles/sfeMovie.dir/src/AudioStream.cpp.obj] Error 1
CMakeFiles\Makefile2:71: recipe for target 'CMakeFiles/sfeMovie.dir/all' failed
mingw32-make[1]: *** [CMakeFiles/sfeMovie.dir/all] Error 2
Makefile:150: recipe for target 'all' failed
mingw32-make: *** [all] Error 2

Maybe you will say i need visual, but this way i compiled it before, except then with shared (dll's) ffmpeg.
I will try that one to with the libass, but:

What version of libass should i use? To (cross) compile them myself, or ready fixed, doesn't matter.
I don't see any others with the same problem.

So, can you help me out?

**Comment by Ceylo** [comment]
I don't know if there are versions of libass that are not supported. I just tried locally with libass 0.14.0 and it built fine. I installed it using Homebrew on macOS.

Regarding Windows builds, sfeMovie build system only supports Visual Studio, so you're inviting yourself to issues with MinGW/cross compilation and I won't help on that part.

**Comment by crazybytes** [comment]
Yeah, ok. Somehow it compiles with the standard mingw32 lib from codeblocks, after compiling the ffmpeg-2.8 manually with native bash under windows.
But no when installed ubuntu shell, i can not get a native bash.exe shell running the standard script.
When compiling 32 bits shared dll files with ffmpeg, the codeblocks mingw somehow manage to deliver good ffmpeg libs, dll's, sfemovielib, that goes with static sfml libs.
But its a jungle. I understand the complexity of all code and scripts etc., but i cannot understand there are people who can get along with all the compilers, settings, cross compilers, shells, msys2, mingw32 mingw-64, mingw under msys, 686, 86_64, host, target, etc. etc. etc.
Seems to me like a setup. No guideline works as they show.
